flockoff/validator/database.py

```
@@ -1,57 +1,123 @@
+            c.execute(
+                """
+                INSERT INTO dataset_revisions(namespace, revision)
+                VALUES (?, ?)
+                ON CONFLICT(namespace) DO UPDATE SET revision=excluded.revision
+                """,
+                (namespace, revision),
+            )
+            self.conn.commit()
+        except sqlite3.Error as e:
+            logger.error(f"Failed to set revision for namespace {namespace}: {str(e)}")
+            raise DatabaseError(f"Failed to update revision: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)